### PR TITLE
Enable and fix `gomoddirectives`, `gomodguard`, `noctx` and `wastedassign` linters

### PR DESCRIPTION
**Pull request by saschagrunert**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

Enabling the linters and fixing the reports.

#### Which issue(s) this PR fixes:

None

#### Special notes for your reviewer:
None
#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

File: internal/factory/container/container.go [hunk 1]
```diff
@@ -552,13 +552,11 @@ func (c *container) AddUnifiedResourcesFromAnnotations(annotationsMap map[string
...
 		// if the value is not specified in base64, then use its raw value.
-		v := ""
 		if err == nil {
-			v = string(d)
+			c.spec.Config.Linux.Resources.Unified[parts[0]] = string(d)
 		} else {
-			v = parts[1]
+			c.spec.Config.Linux.Resources.Unified[parts[0]] = parts[1]
 		}
-		c.spec.Config.Linux.Resources.Unified[parts[0]] = v
```
Comment (kwilczynski) [inline internal/factory/container/container.go:561]: This linter suggestion is an interesting one. The original code does not look all that bad, definitely more readable.

Comment (saschagrunert) [reply internal/factory/container/container.go:561]: That was more or less my decision to get rid of `v`. Do you want to have it back as `var v string`?

Comment (kwilczynski) [reply internal/factory/container/container.go:561]: No worries. Personally, I find the old version slightly easier to read, let it be "easier on the eyes", so to speak. But, we can keep the new one, if that is your preference. :smile:

**Pull request merged**